### PR TITLE
scripts: Eliminate CI variable as it is not longer used

### PR DESCRIPTION
**Pull request by GabyCT**
This PR removes the CI variable which is not longer being used or valid in the kata containers repository. The CI variable was used when we were using jenkins and scripts setups which are not longer supported as now we are using GHA.

### CODE REVIEW

**Review by gkurz — COMMENTED** [review]
> **rootfs: Eliminate CI variable as it is not longer used**
> This PR removes the CI variable which is not longer being used or valid
in the kata containers repository.

Hi @GabyCT,

I don't quite understand why the CI variable is no longer valid. What about all the other users of the CI variable ?

```
$ git grep -E '\$\{?CI[:}"]'
docs/use-cases/using-Intel-QAT-and-kata.md:$ sudo -E make all -j $($(nproc ${CI:+--ignore 1}))
docs/use-cases/using-Intel-QAT-and-kata.md:$ sudo -E make INSTALL_MOD_PATH=$ROOTFS_DIR qat-driver-install -j $($(nproc ${CI:+--ignore 1}))
tests/functional/tracing/test-agent-shutdown.sh:                if [ "${CI:-}" = "true" ]
tests/functional/tracing/tracing-test.sh:       if [ -n "${CI:-}" ]; then
tools/osbuilder/dockerfiles/QAT/run.sh:    make all -j $($(nproc ${CI:+--ignore 1})) 
tools/osbuilder/tests/test_config.sh:if [ -n "${CI:-}" ]; then
tools/osbuilder/tests/test_images.sh:readonly CI=${CI:-}
tools/osbuilder/tests/test_images.sh:   if [ -n "$CI" ]; then
tools/osbuilder/tests/test_images.sh:   if [ -z "$CI" ]; then
tools/osbuilder/tests/test_images.sh:                   [ -n "$CI" ] && sudo -E touch "${ci_results_dir}/${d}_fail"
tools/osbuilder/tests/test_images.sh:                   [ -n "$CI" ] && touch "${ci_results_dir}/${d}_fail"
tools/packaging/kata-deploy/local-build/Makefile:       ${MAKE} -f $(MK_PATH) all -j $(shell nproc ${CI:+--ignore 1}) V=
tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh:CI="${CI:-}"
tools/packaging/kata-deploy/local-build/kata-deploy-binaries-in-docker.sh:      --env CI="${CI}" \
tools/packaging/kernel/build-kernel.sh: make -j $(nproc ${CI:+--ignore 1}) ARCH="${arch_target}" ${CROSS_BUILD_ARG}
tools/packaging/kernel/build-kernel.sh:         make -j $(nproc ${CI:+--ignore 1}) INSTALL_MOD_STRIP=1 INSTALL_MOD_PATH=${kernel_path} modules_install
tools/packaging/kernel/build-kernel.sh:         make -j $(nproc ${CI:+--ignore 1}) bindeb-pkg ARCH="${arch_target}"
tools/packaging/kernel/build-kernel.sh:         make -j $(nproc ${CI:+--ignore 1}) rpm-pkg ARCH="${arch_target}"

```

**Comment by GabyCT** [comment]
@gkurz those need to be removed as well CI variable was used when we were using the building scripts on the kata containers tests repo (setup.sh and so on) that is not longer being used as we do not support a CI setup script where we used CI variable

**Comment by sprt** [comment]
I think we should adapt the files @gkurz pointed out as well then.

**Comment by GabyCT** [comment]
@sprt @gkurz changes applied

File: tools/osbuilder/tests/test_config.sh [hunk 1]
```diff
@@ -34,30 +34,28 @@ distro_in_set() {
...
 
-if [ -n "${CI:-}" ]; then
-	# Since too many distros timeout for now, we only test clearlinux and ubuntu. We can enable other distros when we fix timeout problem.
-	for distro in "${distros[@]}"; do
-		if distro_in_set "${distro}" "${test_distros[@]}"; then
-			continue
-		fi
-		skipWhenTestingAll+=("${distro}")
-	done
+# Since too many distros timeout for now, we only test clearlinux and ubuntu. We can enable other distros when we fix timeout problem.
```
Comment (gkurz) [inline tools/osbuilder/tests/test_config.sh:37]: It seems wrong to keep this code that is supposed to only be run when CI is set. Can you clarify @GabyCT ?

Comment (GabyCT) [reply tools/osbuilder/tests/test_config.sh]: @gkurz @sprt thanks for the feedback changes applied, sorry that I missed that

File: tools/osbuilder/tests/test_config.sh [hunk 2]
```diff
@@ -33,29 +33,3 @@ distro_in_set() {
 	done
 	return 1
 }
-
```
Comment (gkurz) [inline tools/osbuilder/tests/test_config.sh:36]: Wrong commit... this hunk should go to `tools: Remove CI variable in test config osbuilder script`.

Comment (GabyCT) [reply tools/osbuilder/tests/test_config.sh:36]: done

**Review by gkurz — APPROVED** [review]
I've squashed the revert and new fix to `tools/osbuilder/tests/test_config.sh` into the original commit for a prettier `git log`.

Thanks @GabyCT !

**Comment by gkurz** [comment]
Added my SoB to silent the DCO check.

**Review by jepio — APPROVED** [review]
lgtm

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**